lib/membership/events.js

```diff
@@ -20,13 +20,24 @@
 'use strict';
 
 function LocalMemberLeaveEvent(member, oldStatus) {
-    this.name = LocalMemberLeaveEvent.Name;
+    this.name = this.constructor.name;
     this.member = member;
     this.oldStatus = oldStatus;
 }
 
-LocalMemberLeaveEvent.Name = 'localMemberLeave';
+function ReusableEvent(member, oldDampScore) {
```

Comment:
Reusable is not immediately obvious, particularly within the context of this PR because it's only ever fired, but not acted upon. I will add a comment. Good idea.

Re: the prefix. Maybe. I don't want to be overly verbose. Lemme chew.